File: ParteC#/ProyectoProgramacion3/ProyectoProgramacion3Model/Model/reportes/ReporteVentas.cs
using ProyectoProgramacion3Model.Model.reservas;

namespace ProyectoProgramacion3Model.Model.reportes;

public class ReporteVentas : Reporte
{
    public List<Reserva> detalleVentas { get; set; } = new List<Reserva>();
    public int totalVentas { get; set; }
    public double montoTotalGenerado { get; set; }
    public ReporteVentas(int idReporte, DateOnly fechaGeneracion, DateOnly fechaInicioFiltro, DateOnly fechaFinFiltro, List<Reserva> detalleVentas, int totalVentas, double montoTotalGenerado)
        : base(idReporte, fechaGeneracion, fechaInicioFiltro, fechaFinFiltro)
    {
        this.detalleVentas = detalleVentas;
        this.totalVentas = totalVentas;
        this.montoTotalGenerado = montoTotalGenerado;
    }
    public ReporteVentas()
    {
    }
}

File: Backend/ProyectoPro3Modelo/src/main/java/pe/edu/pucp/proyectopro3/modelo/reportes/ReporteVentas.java
package pe.edu.pucp.proyectopro3.modelo.reportes;

import pe.edu.pucp.proyectopro3.modelo.reservas.Reserva;

import java.util.Date;
import java.util.List;

public class ReporteVentas extends Reporte {
    private List<Reserva> detalleVentas;
    private int totalVentas;
    private double montoTotalGenerado;

    public ReporteVentas(int idReporte, Date fechaGeneracion, Date fechaInicioFiltro, Date fechaFinFiltro, List<Reserva> detalleVentas, int totalVentas, double montoTotalGenerado) {
        super(idReporte, fechaGeneracion, fechaInicioFiltro, fechaFinFiltro);
        this.detalleVentas = detalleVentas;
        this.totalVentas = totalVentas;
        this.montoTotalGenerado = montoTotalGenerado;
    }

    public ReporteVentas() {
        super();
    }

    public List<Reserva> getDetalleVentas() {
        return detalleVentas;
    }

    public void setDetalleVentas(List<Reserva> detalleVentas) {
        this.detalleVentas = detalleVentas;
    }

    public int getTotalVentas() {
        return totalVentas;
    }

    public void setTotalVentas(int totalVentas) {
        this.totalVentas = totalVentas;
    }

    public double getMontoTotalGenerado() {
        return montoTotalGenerado;
    }

    public void setMontoTotalGenerado(double montoTotalGenerado) {
        this.montoTotalGenerado = montoTotalGenerado;
    }
}
